### PR TITLE
CU-mz87nx: unit tests for maker-otc matching market

### PR DESCRIPTION
**Pull request by najienka**

### CODE REVIEW

**Comment by melks** [comment]
Task linked: [CU-mz87nx Write functional tests for existing MakerOTC Matching market](https://app.clickup.com/t/mz87nx)

**Review by hiddentao — CHANGES_REQUESTED** [review]
Looking good.

However I've noticed that test cases have a dependency on each other. This makes the test suite unnecessarily fragile, since changing a single test case will require consideration of how it impacts other test cases.

In our existing test suite all test cases are independent of each other - we make use of ganache's snapshot capability (example)[https://github.com/nayms/contracts/blob/master/test/policyClaims.js#L273-L279] as well as nested `describe()` blocks to progressively setup test data as needed (example)[https://github.com/nayms/contracts/blob/master/test/policyClaims.js#L746].

Please refactor the test suite in this manner so that test cases are independent.

File: test/matchingMarket.js [hunk 1]
```diff
@@ -0,0 +1,785 @@
...
+                true,
+                {from: accounts[2]}
+            )
+        })
+
+        it('get correct last offer id after creating offers', async () => {
+            await matchingMarketInstance.last_offer_id().should.eventually.eq(2)
+        })
+
+        it('should not match all three prior offers if the prices do not match', async () => {
```
Comment (hiddentao) [inline test/matchingMarket.js:540]: match all three?

File: test/matchingMarket.js [hunk 2]
```diff
@@ -0,0 +1,785 @@
...
+            await matchingMarketInstance.getOffer(4)
+            .should.eventually.matchObj({
+                '0': toBN(0), // previously toBN(5e18)
+                '1': ADDRESS_ZERO, // previously WETH
+                '2': toBN(0), // previously toBN(10e18)
+                '3': ADDRESS_ZERO // previously DAI
+            })
+
+            await matchingMarketInstance.last_offer_id().should.eventually.eq(3)
+        })
```
Comment (hiddentao) [inline test/matchingMarket.js:635]: Pls add a final account balance check

File: test/matchingMarket.js [hunk 3]
```diff
@@ -0,0 +1,785 @@
...
+            })
+
+            await matchingMarketInstance.getOffer(2)
+            .should.eventually.matchObj({
+                '0': toBN(5e18),
+                '1': erc20WETH.address, 
+                '2': toBN(10e18), 
+                '3': erc20DAI.address 
+            })
+        })
```
Comment (hiddentao) [inline test/matchingMarket.js:701]: Pls add a final account balance check

File: test/matchingMarket.js [hunk 4]
```diff
@@ -0,0 +1,785 @@
...
+            })
+
+            await matchingMarketInstance.getOffer(2)
+            .should.eventually.matchObj({
+                '0': toBN(0),
+                '1': ADDRESS_ZERO, 
+                '2': toBN(0), 
+                '3': ADDRESS_ZERO 
+            })
+        })
```
Comment (hiddentao) [inline test/matchingMarket.js:772]: Pls add a final account balance check

**Review by hiddentao — CHANGES_REQUESTED** [review]
Looking good!

File: test/matchingMarket.js (hunk 4 above)
Comment (najienka) [reply test/matchingMarket.js:772]: resolved

File: test/matchingMarket.js (hunk 3 above)
Comment (najienka) [reply test/matchingMarket.js:701]: resolved

File: test/matchingMarket.js (hunk 2 above)
Comment (najienka) [reply test/matchingMarket.js:635]: resolved

File: test/matchingMarket.js (hunk 1 above)
Comment (najienka) [reply test/matchingMarket.js:540]: typo. Apologies

File: test/matchingMarket.js [hunk 5]
```diff
@@ -0,0 +1,785 @@
...
+
+            await matchingMarketInstance.getOffer(1)
+            .should.eventually.matchObj({
+                '0': toBN(20e18),
+                '1': erc20DAI.address,
+                '2': toBN(10e18),
+                '3': erc20WETH.address
+            })
+
+            await matchingMarketInstance.sellAllAmount(erc20WETH.address, toBN(15e18), erc20DAI.address, toBN(30e18), {from: accounts[1]}).should.be.rejectedWith('revert')
```
[the remark this answers is not in the document]
Comment (najienka) [reply test/matchingMarket.js:378]: resolved

File: package.json [hunk 6]
```diff
@@ -47,6 +47,7 @@
     "@commitlint/cli": "^12.1.1",
     "@commitlint/config-conventional": "^12.1.1",
     "@gnosis.pm/safe-contracts": "^1.2.0",
+    "@openzeppelin/test-helpers": "^0.5.11",
```
Comment (hiddentao) [inline package.json:50]: is this still used?

Comment (najienka) [reply package.json:50]: Thanks for spotting this!

**Comment by najienka** [comment]
Unit tests for policy are failing in this PR build but passing in master branch. Code has been merged in a new branch for new PR.

Closing PR.

**Comment by najienka** [comment]
Unit tests for policy are failing in this PR build but passing in master branch. Code has been merged in a new branch for new PR.

Closing PR.

**Pull request closed without merging**